hal/src/nRF52840/ota_flash_hal.cpp

```diff
@@ -187,13 +187,10 @@ void HAL_System_Info(hal_system_info_t* info, bool construct, void* reserved)
 #ifndef MODULAR_FIRMWARE
 #error HYBRID_BUILD must be modular
 #endif
-                static module_info_t hybrid_info;
                 // change monolithic firmware to modular in the hybrid build.
-                if (!hybrid_module_found && info->modules[i].info->module_function == MODULE_FUNCTION_MONO_FIRMWARE) {
-                    memcpy(&hybrid_info, info->modules[i].info, sizeof(hybrid_info));
-                    info->modules[i].info = &hybrid_info;
-                    hybrid_info.module_function = MODULE_FUNCTION_SYSTEM_PART;
-                    hybrid_info.module_index = 1; 
+                if (!hybrid_module_found && info->modules[i].info.module_function == MODULE_FUNCTION_MONO_FIRMWARE) {
```

Comment:
The `HAL_OTA_Add_System_Info()` will add additional info, except for the module info. So we cannot reply on `fetch_module()`.

Reply:
That's exactly what I'm saying though that it can continue adding additional info like ICCID and other, and we can now let `fetch_module()` deal with modules.

Reply:
The system properties that is added in `HAL_OTA_Add_System_Info()` is not used for module validation. Besides, to add the system properties we need to allocate memory from heap, while the `fetch_module()` doesn't intend to free that memory going further. `HAL_System_Info()` is designed to allocate and free that memory properly. So I'd like leave it as-is and be consistent with the logic on Gen2.

Reply:
Sorry, @XuGuohui, I'm still confused. I'm not sure what consistency we want to keep with Gen 2. I'm proposing we continue to let `HAL_OTA_Add_System_Info()` deal only with system properties that are appended to the system info (ICCID etc), which is exactly what happens on Gen 2 and now that we have proper support for "special" module types (SoftDevice and NCP) in `fetch_module()` let that handle the module info specifically and not delegate that functionality (calling `platform_radio_stack_fetch_module_info()` etc in `HAL_OTA_Add_System_Info()`). Can you please clarify your idea for me?

Reply:
Okay, I probably got you. I was thinking you'd like to implement what the `HAL_OTA_Add_System_Info()` is doing in `fetch_module()`. So you actually would like to keep `HAL_OTA_Add_System_Info()` to only add system properties and also not skip the NCP and radio stack bounds in `HAL_System_Info()` and let `fetch_module()` take care of that, right?

Reply:
Yes, exactly, otherwise I feel it's a bit confusing that we have two paths where info for these modules gets filled: one for system info, another in `fetch_module()` for dependencies and other purposes.

Reply:
Updated.